packages/sanity/src/core/hooks/useDocumentOperation.ts

```diff
@@ -23,9 +24,11 @@ export function useDocumentOperation(
    */
   const api = useObservable(observable)!
 
-  return useDocumentOperationWithComlinkHistory({
+  const operationsAPI = useDocumentOperationWithComlinkHistory({
     api,
     docTypeName,
     publishedDocId,
   })
+
+  return useOperationAPIWithErrorHandling(operationsAPI)
```

Comment:
Is this the right place to wrap the api that could throw based on the document limits, or are there other places as well?